File: example-expo/.yalc/react-native-ssl-manager/ios/cli/UseSslPinningModule.swift
import Foundation
import React

/**
 * React Native CLI module for SSL Pinning
 * Uses the shared logic for common functionality
 */
@objc(UseSslPinningModule)
class UseSslPinningModule: NSObject {
    
    @objc
    func setUseSSLPinning(_ usePinning: Bool, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        SharedLogic.setUseSSLPinning(usePinning)
        resolve(nil)
    }
    
    @objc
    func getUseSSLPinning(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        let usePinning = SharedLogic.getUseSSLPinning()
        resolve(usePinning)
    }
    
    @objc
    func initializeSslPinning(_ configJsonString: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        let usePinning = SharedLogic.getUseSSLPinning()
        
        if !usePinning {
            let result: [String: Any] = [
                "message": "SSL Pinning is disabled",
                "isEnabled": false,
                "domains": []
            ]
            resolve(result)
            return
        }
        
        guard let config = SharedLogic.parseSslPinningConfig(configJsonString) else {
            reject("INVALID_CONFIGURATION", "Invalid SSL pinning configuration format", nil)
            return
        }
        
        guard SharedLogic.validateSslPinningConfig(config) else {
            reject("INVALID_CONFIGURATION", "Invalid SSL pinning configuration format", nil)
            return
        }
        
        // Here you would implement the actual SSL pinning logic
        // For now, we'll just return a success response
        guard let sha256Keys = config["sha256Keys"] as? [String: [String]] else {
            reject("INVALID_CONFIGURATION", "Invalid SSL pinning configuration format", nil)
            return
        }
        
        let domains = Array(sha256Keys.keys)
        
        let result: [String: Any] = [
            "message": "SSL Pinning initialized successfully",
            "isEnabled": true,
            "domains": domains
        ]
        
        resolve(result)
    }
    
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }
} 

File: example-expo/.yalc/react-native-ssl-manager/android/src/main/java/com/usesslpinning/cli/UseSslPinningModule.kt
package com.usesslpinning.cli

import com.facebook.react.bridge.*
import com.usesslpinning.SharedLogic
import org.json.JSONObject

/**
 * React Native CLI module for SSL Pinning
 * Uses the shared logic for common functionality
 */
class UseSslPinningModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    
    override fun getName(): String {
        return "UseSslPinning"
    }
    
    @ReactMethod
    fun setUseSSLPinning(usePinning: Boolean, promise: Promise) {
        try {
            SharedLogic.setUseSSLPinning(reactApplicationContext, usePinning)
            promise.resolve(null)
        } catch (e: Exception) {
            promise.reject("SET_SSL_PINNING_ERROR", e.message, e)
        }
    }
    
    @ReactMethod
    fun getUseSSLPinning(promise: Promise) {
        try {
            val usePinning = SharedLogic.getUseSSLPinning(reactApplicationContext)
            promise.resolve(usePinning)
        } catch (e: Exception) {
            promise.reject("GET_SSL_PINNING_ERROR", e.message, e)
        }
    }
    
    @ReactMethod
    fun initializeSslPinning(configJsonString: String, promise: Promise) {
        try {
            val usePinning = SharedLogic.getUseSSLPinning(reactApplicationContext)
            
            if (!usePinning) {
                val result = Arguments.createMap().apply {
                    putString("message", "SSL Pinning is disabled")
                    putBoolean("isEnabled", false)
                    putArray("domains", Arguments.createArray())
                }
                promise.resolve(result)
                return
            }
            
            val config = SharedLogic.parseSslPinningConfig(configJsonString)
            if (config == null || !SharedLogic.validateSslPinningConfig(config)) {
                promise.reject("INVALID_CONFIGURATION", "Invalid SSL pinning configuration format")
                return
            }
            
            // Here you would implement the actual SSL pinning logic
            // For now, we'll just return a success response
            val domains = config.getJSONObject("sha256Keys").keys()
            val domainsArray = Arguments.createArray()
            while (domains.hasNext()) {
                domainsArray.pushString(domains.next())
            }
            
            val result = Arguments.createMap().apply {
                putString("message", "SSL Pinning initialized successfully")
                putBoolean("isEnabled", true)
                putArray("domains", domainsArray)
            }
            
            promise.resolve(result)
        } catch (e: Exception) {
            promise.reject("INITIALIZATION_FAILED", "Failed to initialize SSL Pinning", e)
        }
    }
} 